src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -127,9 +205,14 @@ - (GRPCChannel *)channel {
         args[@GRPC_SSL_TARGET_NAME_OVERRIDE_ARG] = _hostNameOverride;
       }
 
-      _channel = [GRPCChannel secureChannelWithHost:_address
-                                 pathToCertificates:_pathToCertificates
-                                        channelArgs:args];
+      @synchronized(self) {
+        if (_channelCreds == nil) {
+          [self setSSLCredentials:nil withPrivateKey:nil withCertChain:nil error:nil];
+        }
+        _channel = [GRPCChannel secureChannelWithHost:_address
+                                          credentials:_channelCreds
+                                          channelArgs:args];
+      }
```

Comment:
Copying the `channelCreds` pointer to a local pointer creates a race condition with the `grpc_channel_credentials_release` in `setSSLCredentials` above.

Options:
1) If we could use `grpc_channel_credentials_ref`, we could safely copy the channel credentials locally, but `grpc_channel_credentials_ref` is not an exported function from the C-core.  The C-core doesn't export direct ref functions currently, so doesn't seem like the right path.
2) Could add reference counting in GRPCHost to guard the calling of `grpc_channel_credentials_release`
3) Document that `setSSLCredentials` is not  multi-threaded thread safe. (and remove the existing `@synchronized` from it
4) keep the `@synchronized` with the current scope.

2 is just a few more lines of code, 4 is the most simple.  Thoughts? Other ideas?